src/overlays/actors/ovl_En_Tru/z_en_tru.c

```diff
@@ -806,13 +806,13 @@ s32 func_80A875AC(Actor* thisx, PlayState* play) {
 s32 func_80A8777C(Actor* thisx, PlayState* play) {
     EnTru* this = THIS;
     s32 ret = 0;
-    s32 temp_v0 = Message_GetState(&play->msgCtx);
+    s32 temp_v0;
 
-    switch (temp_v0) {
-        case 4:
-        case 5:
+    switch (Message_GetState(&play->msgCtx)) {
+        case TEXT_STATE_CHOICE:
+        case TEXT_STATE_5:
             if (Message_ShouldAdvance(play)) {
-                case 16:
+                case TEXT_STATE_16:
```

Comment:
Perhaps this is out of scope, but I think #912 has a better way of handling this. i.e
```c
case TEXT_STATE_CHOICE:
case TEXT_STATE_5:
    if (!Message_ShouldAdvance(play)) {
        break;
    }
// Fallthrough
case Text_State_16: